library/entity/entity/src/main/java/org/quiltmc/qsl/entity/api/QuiltEntityTypeBuilder.java

```diff
@@ -0,0 +1,477 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity.api;
+
+import com.google.common.collect.ImmutableSet;
+import net.minecraft.block.Block;
+import net.minecraft.entity.*;
+import net.minecraft.entity.attribute.DefaultAttributeContainer;
+import net.minecraft.entity.attribute.DefaultAttributeRegistry;
+import net.minecraft.entity.mob.MobEntity;
+import net.minecraft.world.Heightmap;
+import net.minecraft.world.World;
+import org.quiltmc.qsl.entity.impl.QuiltEntityType;
+
+import java.util.Objects;
+
+/**
+ * Extended version of {@link EntityType.Builder} with added registration for
+ * server-&gt;client entity tracking values.
+ *
+ * @param <T> Entity class
+ */
+public class QuiltEntityTypeBuilder<T extends Entity> {
+	private SpawnGroup spawnGroup;
+	private EntityType.EntityFactory<T> factory;
+	private boolean saveable = true;
+	private boolean summonable = true;
+	private int trackRange = 5;
+	private int trackedUpdateRate = 3;
+	private Boolean forceTrackedVelocityUpdates = null;
+	private boolean fireImmune = false;
+	private boolean spawnableFarFromPlayer;
+	private EntityDimensions dimensions = EntityDimensions.changing(-1.0f, -1.0f);
+	private ImmutableSet<Block> specificSpawnBlocks = ImmutableSet.of();
+
+	protected QuiltEntityTypeBuilder(SpawnGroup spawnGroup, EntityType.EntityFactory<T> factory) {
+		this.spawnGroup = spawnGroup;
+		this.factory = factory;
+		this.spawnableFarFromPlayer = spawnGroup == SpawnGroup.CREATURE || spawnGroup == SpawnGroup.MISC;
+	}
+
+	/**
+	 * Creates an entity type builder.
+	 *
+	 * <p>This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.</p>
```

Comment:
<p></p>? that's weird;
```suggestion
	 * Creates an entity type builder.
	 * <p>
	 * This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.
```